bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/v2/TeaserImplTest.java

```diff
@@ -111,4 +112,36 @@ protected void testEmptyTeaserImageDelegatingToFeaturedImage() {
         Utils.testJSONExport(teaser, Utils.getTestExporterJSONPath(testBase, "teaser20"));
     }
 
+    @Test
+    protected void testImageFromPage_withoutLink() {
+        Teaser teaser = getTeaserUnderTest(TEASER_21);
+        Link link = teaser.getLink();
+        assertNull(link, "link should be null");
+        Utils.testJSONExport(teaser, Utils.getTestExporterJSONPath(testBase, "teaser21"));
+    }
+
+    @Test
+    protected void testImageFromPage_withLink() {
+        Teaser teaser = getTeaserUnderTest(TEASER_22);
+        Utils.testJSONExport(teaser, Utils.getTestExporterJSONPath(testBase, "teaser22"));
+    }
+
+    @Test
+    protected void testImageFromPage_withInvalidLink() {
+        Teaser teaser = getTeaserUnderTest(TEASER_22a);
+        Utils.testJSONExport(teaser, Utils.getTestExporterJSONPath(testBase, "teaser22a"));
+    }
+
+    @Test
+    protected void testImageFromPage_withCta() {
+        Teaser teaser = getTeaserUnderTest(TEASER_23);
+        Utils.testJSONExport(teaser, Utils.getTestExporterJSONPath(testBase, "teaser23"));
+    }
```

Comment:
@jckautzmann - would it be worth adding a case here with multiple CTA, to ensure it takes the first one?

Reply:
@richardhand Addressed in https://github.com/adobe/aem-core-wcm-components/pull/1828/commits/2ed75881aa722610e982211bb46f1d9b7da9aaf7